### PR TITLE
[1.5] Plumb and allow customizing ciphers and tls version

### PR DESCRIPTION
**Pull request by mfojtik**
Ref: https://github.com/openshift/origin/pull/13167

@liggitt PTAL

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
LGTM, [merge]

**Comment by mfojtik** [comment]
[test]

File: pkg/cmd/server/kubernetes/master_config.go [hunk 1]
```diff
@@ -298,6 +299,7 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 	genericConfig.LegacyAPIGroupPrefixes = LegacyAPIGroupPrefixes
 	genericConfig.SecureServingInfo.BindAddress = options.ServingInfo.BindAddress
 	genericConfig.SecureServingInfo.BindNetwork = options.ServingInfo.BindNetwork
```
Comment (liggitt) [inline pkg/cmd/server/kubernetes/master_config.go]: you dropped a line here
```
genericConfig.SecureServingInfo.MinTLSVersion = crypto.TLSVersionOrDie(options.ServingInfo.MinTLSVersion)
```

**Comment by liggitt** [comment]
legitimate failure:
```
--- FAIL: TestTLSOverrides (5.37s)
07:23:06 	tls_test.go:116: Expected success true, got false with TLS version VersionTLS10 dialing master
07:23:06 	tls_test.go:116: Expected success true, got false with TLS version VersionTLS11 dialing master
```

**Comment by mfojtik** [comment]
@liggitt fixed

[test]

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**